sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c

```diff
@@ -100,22 +187,26 @@ bool test_main(void) {
           kDifToggleEnabled));
 
       CHECK_DIF_OK(dif_aon_timer_wakeup_stop(&aon_timer));
-      aon_timer_testutils_watchdog_config(&aon_timer, 0xffffffff, 10, false);
 
       // Enter low power mode.
       LOG_INFO("Issue WFI to enter sleep");
+      aon_timer_testutils_watchdog_config(&aon_timer, UINT32_MAX, 20, false);
       wait_for_interrupt();
     } else {
       LOG_INFO("Watchdog reset");
-      // Check that the previously written retention SRAM data can still
-      // be read successfully.
-      retention_sram_check(false);
+      // Check that the previously written retention SRAM data cannot be read
+      // successfully.
+      retention_sram_check(
+          (check_config_t){.do_write = false, .is_equal = false});
     }
   } else {
     LOG_FATAL("Unexepected reset type detected.");
   }
 
   // Turn off the AON timer hardware completely before exiting.
   aon_timer_testutils_shutdown(&aon_timer);
+  if (kDeviceType == kDeviceSimDV) {
+    CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
+  }
   return true;
```

Comment:
It would be safer to return `true` from the conditional clause of interest above and return `false` here, to avoid any possibility of a falsely passing test.

Reply:
Returning `false` here would make the test always fail with FPGA and Verilator.